src/ImageSharp/Formats/Jpeg/PdfJsPort/Components/PdfJsHuffmanTable.cs

```diff
@@ -1,139 +1,105 @@
 ﻿// Copyright (c) Six Labors and contributors.
 // Licensed under the Apache License, Version 2.0.
 
-using System;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 using SixLabors.ImageSharp.Memory;
 
 namespace SixLabors.ImageSharp.Formats.Jpeg.PdfJsPort.Components
 {
     /// <summary>
     /// Represents a Huffman Table
     /// </summary>
-    internal struct PdfJsHuffmanTable : IDisposable
+    [StructLayout(LayoutKind.Sequential)]
+    internal unsafe struct PdfJsHuffmanTable
     {
-        private BasicArrayBuffer<short> lookahead;
-        private BasicArrayBuffer<short> valOffset;
-        private BasicArrayBuffer<long> maxcode;
-        private IManagedByteBuffer huffval;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="PdfJsHuffmanTable"/> struct.
-        /// </summary>
-        /// <param name="memoryManager">The <see cref="MemoryManager"/> to use for buffer allocations.</param>
-        /// <param name="lengths">The code lengths</param>
-        /// <param name="values">The huffman values</param>
-        public PdfJsHuffmanTable(MemoryManager memoryManager, byte[] lengths, byte[] values)
-        {
-            // TODO: Replace FakeBuffer<T> usages with standard or array orfixed-sized arrays
-            this.lookahead = memoryManager.AllocateFake<short>(256);
-            this.valOffset = memoryManager.AllocateFake<short>(18);
-            this.maxcode = memoryManager.AllocateFake<long>(18);
-
-            using (IBuffer<short> huffsize = memoryManager.Allocate<short>(257))
-            using (IBuffer<short> huffcode = memoryManager.Allocate<short>(257))
-            {
-                GenerateSizeTable(lengths, huffsize.Span);
-                GenerateCodeTable(huffsize.Span, huffcode.Span);
-                GenerateDecoderTables(lengths, huffcode.Span, this.valOffset.Span, this.maxcode.Span);
-                GenerateLookaheadTables(lengths, values, this.lookahead.Span);
-            }
-
-            this.huffval = memoryManager.AllocateManagedByteBuffer(values.Length, true);
-            Buffer.BlockCopy(values, 0, this.huffval.Array, 0, values.Length);
-
-            this.MaxCode = this.maxcode.Array;
-            this.ValOffset = this.valOffset.Array;
-            this.HuffVal = this.huffval.Array;
-            this.Lookahead = this.lookahead.Array;
-        }
-
         /// <summary>
         /// Gets the max code array
         /// </summary>
-        public long[] MaxCode
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get;
-        }
+        public FixedInt64Buffer18 MaxCode;
```

Comment:
@JimBobSquarePants Why do we store `Int64` here? Values are copared against a `short` inside `DecodeHuffman()`.

Reply:
Blindly copied from wherever I got the original Huffman code. If short works great!